cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/ApplicationFilter.java

```diff
@@ -110,4 +111,29 @@ public String toString() {
         '}';
     }
   }
+
+  /**
+   * Returns true if the application version is latest
+   */
+  public static class ApplicationLatestFilter extends ApplicationIdFilter {
+    private final ApplicationLifecycleService applicationLifecycleService;
+
+    public ApplicationLatestFilter(ApplicationLifecycleService applicationLifecycleService) {
+      this.applicationLifecycleService = applicationLifecycleService;
```

Comment:
Is this correct to pass in the applicationLifecycleService?

Reply:
why do we want to pass applicationLifecycleService? Can we pass the latest version directly instead?

Reply:
so I think one more thing is that when we want to use the `latest` or `strictEqual` parameter - we should be making using of the scan-by-index spi we added here `https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java#L312`